### PR TITLE
Refactor configurations with support for old properties

### LINKED ISSUE #2300: Refactor configurations
### Description:

For similar meaning representing strings, several names used with configurations.

Example:
Enable vs Enabled
Url vs URL

It is important implement such that existing products does not break due to this introduced change. 

### Environment details (with versions):
- OS:
- Client:
- Env (Docker/K8s):

---

### PR DESCRIPTION
**Pull request by VirajSalaka**
### Purpose

Following set of configurations are updated.

```
enforcer.restServer.enable -> enforcer.restServer.enabled
adapter.consul.enable -> adapter.consul.enabled
enforcer.jwtGenerator.enable -> enforcer.jwtGenerator.enabled

controlPlane.serviceUrl -> controlPlane.serviceURL
globalAdapter.serviceUrl -> globalAdapter.serviceURL
enforcer.throttling.JmsConnectionProviderURL -> enforcer.throttling.JmsConnectionProviderURL
globalAdapter.OverwriteHostName -> globalAdapter.OverrideHostName
enforcer.throttling.publisher.urlGroup -> enforcer.throttling.publisher.URLGroup
```

### Issues

Fixes #2300 

### Automation tests
 - Unit tests added: No
 - Integration tests added: No

### Tested environments

Not Tested

---
#### Maintainers: Check before merge
- [x] Assigned 'Type' label
- [x] Assigned the project
- [x] Validated respective github issues
- [x] Assigned milestone to the github issue(s)

### CODE REVIEW

File: adapter/config/parser.go [hunk 1]
```diff
@@ -162,3 +164,46 @@ func GetControlPlaneConnectedTenantDomain() string {
...
+	if config.GlobalAdapter.ServiceURLDeprecated != UnassignedAsDeprecated {
+		printDeprecatedWarningLog("globalAdapter.serviceUrl", "globalAdapter.serviceURL")
+		config.GlobalAdapter.ServiceURL = config.GlobalAdapter.ServiceURLDeprecated
+	}
+	if config.Enforcer.Throttling.JmsConnectionProviderURLDeprecated != UnassignedAsDeprecated {
+		printDeprecatedWarningLog("enforcer.throttling.JmsConnectionProviderUrl", "enforcer.throttling.JmsConnectionProviderURL")
+		config.Enforcer.Throttling.JmsConnectionProviderURL = config.Enforcer.Throttling.JmsConnectionProviderURLDeprecated
+	}
+	if config.GlobalAdapter.OverwriteHostName != UnassignedAsDeprecated {
+		printDeprecatedWarningLog("globalAdapter.OverwriteHostName", "globalAdapter.OverrideHostName")
```
Comment (praminda) [inline adapter/config/parser.go:182]: shouldn't we override the value here?
```suggestion
		printDeprecatedWarningLog("globalAdapter.OverwriteHostName", "globalAdapter.OverrideHostName")
		onfig.GlobalAdapter.OverrideHostName = onfig.GlobalAdapter.OverwriteHostName
```

**Review by praminda — APPROVED** [review]

File: adapter/config/parser.go (hunk 1 above)
Comment (VirajSalaka) [reply adapter/config/parser.go:182]: fixed.

File: adapter/config/types.go [hunk 2]
```diff
@@ -395,8 +408,10 @@ type APICtlUser struct {
 
 // ControlPlane struct contains configurations related to the API Manager
 type controlPlane struct {
-	Enabled                    bool
-	ServiceURL                 string `toml:"serviceUrl"`
+	Enabled    bool
+	ServiceURL string
```
Comment (pubudu538) [inline adapter/config/types.go:412]: Formatting issues.

Comment (VirajSalaka) [reply adapter/config/types.go:412]: When there is comment in the middle, this is how VSCode formats it. 
If we change it manually, then the developers would have to keep doing it repeatedly.

**Comment by choreo-cicd** [comment]
[failed] dev-deployment : 20211115.4

**Comment by choreo-cicd** [comment]
[succeeded] dev-deployment : 20211115.12

**Comment by choreo-cicd** [comment]
[failed] stage-deployment : 20211117.2

**Comment by choreo-cicd** [comment]
[succeeded] stage-deployment : 20211117.3

**Comment by choreo-cicd** [comment]
[succeeded] prod-deployment : 20211119.1

**Pull request merged**